ext/ffi/lib.rs

```diff
@@ -811,11 +821,46 @@ fn make_sync_fn<'s>(
       // SAFETY: The pointer will not be deallocated until the function is
       // garbage collected.
       let symbol = unsafe { &*(external.value() as *const Symbol) };
+      let needs_unwrap = match needs_unwrap(symbol.result_type) {
```

Comment:
The bool-like name here is a bit confusing.

Reply:
Also, the `needs_unwrap(symbol.result_type)` call & match and the `is_i64(symbol.result_type)` call can be taken outside the closure, if wanted.

Reply:
that would mean leaking it with the Symbol. V8 function closures cannot capture environment.

Reply:
Ah, okay.

Reply:
This can split apart into multiple branches, with different specialised function pointers passed to the `v8::FunctionTemplate::builder`?

This would be three different paths, for `false == needs_unwrap`, `false == is_i64`, `true == is_i64`.

Reply:
needs_unwrap is cheap, happy to have 3 paths if there is a performance benefit.

Reply:
I would also like to presume that it's cheap, but there's also `args.get(symbol.parameter_types.len() as i32)`, which may be better optimised... another PR can measure differences and consider this - no reason to halt the PR on this.